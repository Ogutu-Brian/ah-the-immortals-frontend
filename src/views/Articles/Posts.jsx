import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Spinner from '../Article/Spinner';
import PostFeed from '../../components/Articles/PostFeed';
import { getPosts } from '../../redux/actions/postActions';
import '../../styles/App.css';



class Posts extends Component {
  componentDidMount() {
    const { getPosts } = this.props;
    getPosts();
  }

  render() {
    const { post: { posts, loading } } = this.props;
    return (
      <div>
        <div className="post-feed">
          <br />
          {
            (!posts || loading)? <Spinner />: <PostFeed posts={posts} />
          }
        </div>
        <div className="row">
          <div className="col" />
          <div className="col">
            <div>
              <ul className="pagination pagination-lg">
                <li className="page-item disabled">
                  <a className="page-link" href="#">&laquo;</a>
                </li>
                <li className="page-item active">
                  <a className="page-link" href="#">1</a>
                </li>
                <li className="page-item">
                  <a className="page-link" href="#">2</a>
                </li>
                <li className="page-item">
                  <a className="page-link" href="#">3</a>
                </li>
                <li className="page-item">
                  <a className="page-link" href="#">4</a>
                </li>
                <li className="page-item">
                  <a className="page-link" href="#">5</a>
                </li>
                <li className="page-item">
                  <a className="page-link" href="#">&raquo;</a>
                </li>
              </ul>
            </div>

          </div>
          <div className="col" />
        </div>
        <br />
        <br />
        <br />
        <br />
      </div>
     
    );
  }
}

Posts.propTypes = {
  getPosts: PropTypes.func.isRequired,
  post: PropTypes.object.isRequired
};

const mapStateToProps = state => ({
  post: state.post,
  errors: state.errors
});

export default connect(mapStateToProps, { getPosts })(Posts);
