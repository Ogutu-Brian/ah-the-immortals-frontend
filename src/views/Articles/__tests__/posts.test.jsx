import React from 'react';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import { createBrowserHistory } from 'history';
import { Router } from 'react-router-dom';
import thunk from 'redux-thunk';
import Posts from '../Posts';

describe('<Posts />', () => {
  const testStore = configureMockStore([thunk]);
  const initialState = {
    post: {
      posts: [], loading: false
    }
  };
  const store = testStore(initialState);
  const history = createBrowserHistory();
  const wrapper = mount(
    <Router history={history}>
      <Provider store={store}>
        <Posts history={history} />
      </Provider>
    </Router>
  );
  it('passes', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
