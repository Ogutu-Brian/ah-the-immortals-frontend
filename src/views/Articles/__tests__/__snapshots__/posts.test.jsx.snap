// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> passes 1`] = `ReactWrapper {}`;
